### ISSUE TITLE
Que upgrade does not set job job_schema_version to 2

### ISSUE THREAD

**Issue by hlascelles** [que-rb/que#429]
We recently upgraded a project from to 2.x and found that all of the extant jobs were not being executed. It appears that the `job_schema_version` was still set to 1. This meant the new workers would ignore the enqueued jobs.

To get all the jobs running, we ran the following migration:

```sql
UPDATE que_jobs
SET job_schema_version = 2
WHERE job_schema_version IS DISTINCT FROM 2
```

I can imagine there is a rationale for this, such as to allow gradual no-downtime migration from one version to another. But it seems to have bitten a few people. Certainly us, a few here https://github.com/hlascelles/que-scheduler/issues/412, and maybe this one https://github.com/que-rb/que/issues/381.

Is there somewhere in the docs that we should put this as a debug step? And/or should it go in the migrate guide here: https://github.com/que-rb/que/blob/c0b0be8dc60ff18ea6a76a553ec38a273c303de3/docs/README.md#migrating

**Comment by hlascelles** [comment]
This is probably related: https://github.com/que-rb/que/pull/365

**Comment by ZimbiX** [comment]
This is by design. Did you miss the upgrade guide? https://github.com/que-rb/que/blob/master/CHANGELOG.md#200beta1-2022-03-24 I guess we assumed people would check the changelog when upgrading a major version

**Comment by ZimbiX** [comment]
Maybe we could add a startup check that if any jobs exist with job_schema_version of 1 then a warning is logged with a link to the upgrade guide.

**Comment by hlascelles** [comment]
Yes, looks like we (and a few others) missed it. Hmm, maybe it should be copied into the main docs. 

I see the [Migrating guide](https://github.com/que-rb/que/blob/c0b0be8dc60ff18ea6a76a553ec38a273c303de3/docs/README.md#migrating). Perhaps it can go in or near there?

And yes, a warning on startup sounds ideal!

**Comment by hlascelles** [comment]
Alternatively we can close this and just leave it as a documentation pointer. Many people are probably now upgraded. Up to you. :+1:
